c8/tenant.py

```diff
@@ -584,6 +587,103 @@ def response_handler(resp):
 
         return self._execute(request, response_handler)
 
+    def save_query(self, query_data):
+        """Save query.
+
+        :param query_data: query data
+        :type query_data: dict
+        :return: Results of saved query
+        :rtype: dict
+        :raise c8.exceptions.SavedQueriesPostError: if query creation failed
+        """
+
+        request = Request(method="post", endpoint="/userqueries",
+                          data=query_data)
+
+        def response_handler(resp):
+            if not resp.is_success:
+                raise SavedQueriesCreateError(resp, request)
+            return resp.body["result"]
+
+        return self._execute(request, response_handler)
+
+    def execute_saved_query(self, query_name):
+        """Execute saved query.
+
+        :param query_name: query name
+        :type query_name: str | unicode
+        :return: Results of execute saved query
+        :rtype: dict
+        :raise c8.exceptions.SavedQueriesPostError: if query creation failed
+        """
+
+        request = Request(method="post",
+                          endpoint="/userqueries/execute/%s" % query_name)
+
+        def response_handler(resp):
+            if not resp.is_success:
+                raise SavedQueriesExecuteError(resp, request)
+            return resp.body
+
+        return self._execute(request, response_handler)
+
+    def get_saved_queries(self):
+        """Get all saved query.
+
+        :return: Details of all saved queries
+        :rtype: list
+        :raise c8.exceptions.SavedQueriesGetError: if getting query failed
+        """
+
+        request = Request(method="get", endpoint="/userqueries/user")
+
+        def response_handler(resp):
+            if not resp.is_success:
+                raise SavedQueriesListError(resp, request)
+            return resp.body["result"]
+
+        return self._execute(request, response_handler)
+
+    def update_saved_query(self, query_name, query_data):
+        """Update saved query.
+
+        :param query_name: name of query
+        :type query_name: str | unicode
+        :param query_data: query data
+        :type query_data: dict
+        :return: True if query is deleted
+        :rtype: bool
+        :raise c8.exceptions.SavedQueriesUpdateError: if query update failed
+        """
+        request = Request(method="patch", data=query_data,
```

Comment:
It is a `put` call. Is the `method = patch` correct here?

Reply:
updated it